zipline/data/minute_bars.py

```diff
@@ -546,7 +553,51 @@ def write_cols(self, sid, dts, cols):
                     len(dts),
                     " ".join("{0}={1}".format(name, len(cols[name]))
                              for name in self.COL_NAMES)))
-        self._write_cols(sid, dts, cols)
+        if overwrite:
+            self._overwrite_cols(sid, dts, cols)
+        else:
+            self._write_cols(sid, dts, cols)
+
+    def _overwrite_cols(self, sid, dts, cols):
+        """
+        Internal method for `write_cols` and `write`. Overwrite data for the
+        given sid and cols on the given dts
+
+        Parameters:
+        -----------
+        sid : int
+            The asset identifier for the data being overwritten.
+        dts : datetime64 array
+            The dts corresponding to values in cols.
+        cols : dict of str -> np.array
+            dict of market data with the following characteristics.
+            keys are ('open', 'high', 'low', 'close', 'volume')
+            open : float64
+            high : float64
+            low  : float64
+            close : float64
+            volume : float64|int64
+        """
+        sidpath = self.sidpath(sid)
+        if not os.path.exists(sidpath):
+            raise BcolzTableDoesNotExist('Could not find bcolz table for sid '
+                                         '%s while trying to overwrite data'
+                                         % sid)
```

Comment:
`self._ensure_ctable` will give us a blank table if one doesn't exist for the path. Although that is unlikely to happen in the context for which this method is currently used, I'd rather it just raise an error if it does happen (I don't want to be trying to overwrite parts of an empty table)